### DISCUSSION TITLE
How do rewrites affect SEO

### DISCUSSION THREAD

**Discussion by ljrodriguez1** [vercel/community#1841] [Help] [answered]
Hi, I have been reading several blog posts about what is the best way to have a blog to boost SEO on your page, one of the points is the URL of the blog which could be one of this 2 things:
- blog.example.com (sub-domain)
- example.com/blog (sub-page)
Every post states it is better to have it as a sub-page because google can associate it with your main page.

The problem is that I don't want to have the same vercel project, for the blog and my main page, and a solution I found was [vercel rewrites](https://vercel.com/docs/concepts/projects/project-configuration#rewrites).

My question is, will this have the best of both worlds, good SEO and separate projects, or will google consider that the blog is still in a sub-domain.

If the latter is correct, how can I have different project/repositories/frameworks for subpages, where google still consider them as the same domain for SEO purposes.

**Comment by amyegan** [comment, marked as answer]
Good question, Lucas. The main concern here would be duplicate content, and there are a couple of ways to handle that. One option is to configure your rewrite such that a rewritten page can only be accessed by one URL. We have a guide that can help with setting up the rewrite: [Can I redirect from a subdomain to a subpath?](https://vercel.com/guides/can-i-redirect-from-a-subdomain-to-a-subpath)

 The other option is to specify [canonical URLs](https://developers.google.com/search/docs/crawling-indexing/consolidate-duplicate-urls#define-canonical) which enables you to allow access at either URL while letting search engines know which one is the source of truth

**Comment by salteood789** [comment]
Rewrites can have both positive and negative effects on SEO, depending on how they are implemented. If a rewrite improves the quality and relevance of the content, it can help to improve SEO rankings. On the other hand, if a rewrite is done poorly and results in a loss of quality or relevance, it can hurt SEO rankings.

For example, if you take a piece of content that is poorly written and re-write it to improve its quality, readability, and relevance to a specific topic, it may perform better in search engine rankings. In contrast, if you take a well-written piece of content and re-write it in a way that makes it less relevant or useful to the intended audience, it could negatively impact its SEO performance.

parentsqueries.com/babysitting-twins-fair-pay, as an example, let's say the original article was poorly written and didn't provide much value to the reader. If the author were to rewrite the article, improving the quality and relevance of the content, it could potentially improve its SEO performance.

For example, the author could conduct more research and provide more comprehensive information on how to determine fair pay for babysitting twins. They could also optimize the article with relevant keywords to help it rank higher in search engine results pages for related queries. If the rewrite improves the article's quality and relevance, it may attract more traffic and links, further boosting its SEO performance over time.

**Reply by amyegan** [reply]
Hi @salteood789! We were talking about [URL rewrites](https://vercel.com/docs/concepts/projects/project-configuration#rewrites), not content rewrites. But you're right that it is also worth considering page content when talking about SEO

**Comment by JKarlavige** [comment]
Hi @amyegan,

We are working on incrementally migrating our marketing site from Jekyll to NextJS using the `fallback` rewrite method noted in [the docs](https://nextjs.org/docs/pages/api-reference/next-config-js/rewrites#incremental-adoption-of-nextjs). This rewrite is setup in our new NextJS version of the site hosted on Vercel. 

I came across this discussion as our main concern is negatively affecting SEO. 

The rewrite looks to work correctly. If a NextJS page is found for the current route, it shows that page. Otherwise, it falls back to our page on Jekyll, which we have on a separate URL. Our rewrite setup looks like the following:

```
async rewrites() {
    return {
      fallback: [
        {
          source: '/:path*',
          destination: `old.www.<domain>.com/:path*`,
        },
      ],
    }
  },
```

However, since we have our old Jekyll site living on a separate URL, the site is available by accessing this temporary URL directly. I tried setting up redirects to redirect any requests from the temporary URL to the production URL, however it looks like that creates an infinite loop and breaks our rewrites.

Are canonicals our best option here to tell search engines which URL is our production site? Or is there another way we can prevent the old domain from being accessible with a rewrite/redirect?

For additional context, the old site is currently hosted on Netlify. Thank you!
